test/lit/passes/outlining.wast

```diff
@@ -49,60 +59,152 @@
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 2)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+
+  ;; CHECK:      (func $c
   ;; CHECK-NEXT:  (drop
   ;; CHECK-NEXT:   (i32.const 7)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT:  (call $outline$
   ;; CHECK-NEXT:   (i32.const 4)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT: )
-  (func $a
+  (func $c
     (drop (i32.const 7))
     (drop (i32.const 4))
     (drop (i32.const 1))
     (drop (i32.const 2))
   )
-  ;; CHECK:      (func $b
+  ;; CHECK:      (func $d
   ;; CHECK-NEXT:  (drop
   ;; CHECK-NEXT:   (i32.const 0)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT:  (call $outline$
   ;; CHECK-NEXT:   (i32.const 5)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT: )
-  (func $b
+  (func $d
     (drop (i32.const 0))
     (drop (i32.const 5))
     (drop (i32.const 1))
     (drop (i32.const 2))
   )
 )
-;; CHECK:      (func $outline$ (param $0 i32)
-;; CHECK-NEXT:  (drop
-;; CHECK-NEXT:   (local.get $0)
-;; CHECK-NEXT:  )
+
+;; Tests that outlining occurs properly when the sequence is at the beginning of a function.
+;; Also tests that the outlined function has no arguments.
+(module
+  ;; CHECK:      (type $0 (func))
+
+  ;; CHECK:      (func $outline$
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 0)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.add
+  ;; CHECK-NEXT:    (i32.const 0)
+  ;; CHECK-NEXT:    (i32.const 1)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+
+  ;; CHECK:      (func $e
+  ;; CHECK-NEXT:  (call $outline$)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 6)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $e
+    (drop (i32.const 0))
+    (drop (i32.add
+      (i32.const 0)
+      (i32.const 1)))
+    (drop (i32.const 1))
+    (drop (i32.const 6))
+  )
+  ;; CHECK:      (func $f
+  ;; CHECK-NEXT:  (call $outline$)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 7)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $f
+    (drop (i32.const 0))
+    (drop (i32.add
+      (i32.const 0)
+      (i32.const 1)))
+    (drop (i32.const 1))
+    (drop (i32.const 7))
+  )
+)
+
+;; Tests multiple sequences being outlined from the same source function into different
+;; outlined functions.
+(module
+;; CHECK:      (type $0 (func))
+
+;; CHECK:      (func $outline$_4
 ;; CHECK-NEXT:  (drop
-;; CHECK-NEXT:   (i32.const 1)
+;; CHECK-NEXT:   (i32.sub
+;; CHECK-NEXT:    (i32.const 3)
+;; CHECK-NEXT:    (i32.const 4)
+;; CHECK-NEXT:   )
 ;; CHECK-NEXT:  )
+;; CHECK-NEXT: )
+
+;; CHECK:      (func $outline$
 ;; CHECK-NEXT:  (drop
-;; CHECK-NEXT:   (i32.const 2)
+;; CHECK-NEXT:   (i32.add
+;; CHECK-NEXT:    (i32.const 0)
+;; CHECK-NEXT:    (i32.const 1)
+;; CHECK-NEXT:   )
 ;; CHECK-NEXT:  )
 ;; CHECK-NEXT: )
+
+;; CHECK:      (func $g
+;; CHECK-NEXT:  (call $outline$)
+;; CHECK-NEXT:  (call $outline$_4)
+;; CHECK-NEXT: )
+  (func $g
```

Comment:
Is this the most recent output of the update script? Usually it lines up the checks with the functions.

Reply:
done